derrida/interventions/management/commands/annotation_data.py

```diff
@@ -68,13 +67,39 @@ def handle(self, *args, **kwargs):
             for intervention in data:
                 csvwriter.writerow(self.flatten_dict(intervention))
 
-    def intervention_data(self, intervention):
+    def annotation_data(self, intervention):
         '''Generate a dictionary of data to export for a single
         :class:`~derrida.books.models.Reference` object'''
 
         # NOTE: using OrderedDict to ensure JSON output follows logical
         # order in Python < 3.6, where dict order is not guaranteed
 
+        if intervention.canvas:
+            canvas_url = reverse('books:canvas-image', kwargs={
+                'slug': intervention.work_instance.slug,
+                'short_id': intervention.canvas.short_id, 
+                'mode': 'large',
+            })
+            iiif_base = ('https://derridas-margins.princeton.edu' + canvas_url).replace('/large/', '/iiif/')
+            page_iiif = iiif_base + 'full/500,/0/default.jpg'
+        else:
+            page_iiif = ''
+
+        # This assumes that any annotation would have a canvas and iiif_base has
+        #  already been set.
+        if intervention.iiif_image_selection():
+            # `canonicalize` makes a request for each item. While iterating, 
+            #  it may be worth commenting out that method.
+            iiif_query = re.sub(
+                r'https://iiif-cloud.*intermediate_file/', '', 
+                str(intervention.iiif_image_selection().canonicalize())
+            )
+            annotation_region = iiif_base + iiif_query
```

Comment:
This should be nested under the other `if` for an associated canvas, since it's impossible without an associated canvas and we need those variables.

And the image selection should use the iiif image object constructed above – either by setting the api endpoint & id on the selection version based on the first one, or by copying the region from this one into the first one. Probably better to call canonicalize before changing the endpoint, though!

Reply:
✅ 49244c4